File: src/main/java/net/teamhollow/direbats/client/render/entity/DirebatFangArrowEntityRenderer.java
package net.teamhollow.direbats.client.render.entity;

import net.fabricmc.api.EnvType;
import net.fabricmc.api.Environment;
import net.minecraft.client.render.entity.EntityRendererFactory;
import net.minecraft.client.render.entity.ProjectileEntityRenderer;
import net.minecraft.entity.projectile.PersistentProjectileEntity;
import net.minecraft.util.Identifier;

import static net.teamhollow.direbats.client.util.ClientUtil.entityTexture;

@Environment(EnvType.CLIENT)
public class DirebatFangArrowEntityRenderer<T extends PersistentProjectileEntity> extends ProjectileEntityRenderer<T> {
    public static final Identifier TEXTURE = entityTexture("arrow/direbat_fang_arrow");

    public DirebatFangArrowEntityRenderer(EntityRendererFactory.Context ctx) {
        super(ctx);
    }

    @Override
    public Identifier getTexture(T entity) {
        return TEXTURE;
    }
}

File: src/main/kotlin/dev/andante/direbats/client/render/entity/DirebatFangArrowEntityRenderer.kt
package dev.andante.direbats.client.render.entity

import dev.andante.direbats.Direbats
import dev.andante.direbats.entity.DirebatFangArrowEntity
import net.fabricmc.api.EnvType
import net.fabricmc.api.Environment
import net.minecraft.client.render.entity.EntityRendererFactory
import net.minecraft.client.render.entity.ProjectileEntityRenderer
import net.minecraft.entity.projectile.PersistentProjectileEntity
import net.minecraft.util.Identifier

/**
 * Represents the renderer for a [DirebatFangArrowEntity].
 */
@Environment(EnvType.CLIENT)
class DirebatFangArrowEntityRenderer<T : PersistentProjectileEntity>(ctx: EntityRendererFactory.Context) : ProjectileEntityRenderer<T>(ctx) {
    override fun getTexture(entity: T): Identifier {
        return TEXTURE
    }

    companion object {
        val TEXTURE = Identifier(Direbats.MOD_ID, "textures/entity/arrow/direbat_fang_arrow.png")
    }
}
